test/test_torch.py

```diff
@@ -7045,6 +7045,15 @@ def test_cat_out_channels_last(self, device):
         res2 = torch.cat((x, y), out=z)
         self.assertEqual(res1, res2)
 
+    def test_cat_in_channels_last(self, device):
+        x = torch.randn((4, 15, 8, 8))
+        y = torch.randn(x.shape)
+        res1 = torch.cat((x, y), dim=1)
+        x = x.clone().contiguous(memory_format=torch.channels_last)
+        y = y.clone().contiguous(memory_format=torch.channels_last)
+        res2 = torch.cat((x, y), dim=1)
+        self.assertEqual(res1, res2)
```

Comment:
Please check format of the output

Reply:
Please add test covering both variants when `result.numel() < at::internal::GRAIN_SIZE` and when it is not. 
Please add tests covering various `dim=` values.

Reply:
Sounds good. Will do.